### PR TITLE
Fixed some spelling and removed an unused function 

### PR DESCRIPTION
**Pull request by levigross**
that could be problematic. `url.Parse` fails with a single `%` character and would panic unnecessarily
 
Signed-off-by: Levi Gross <levi@levigross.com>

### CODE REVIEW

File: registry/api/v2/urls.go [hunk 1]
```diff
@@ -220,15 +220,3 @@ func appendValuesURL(u *url.URL, values ...url.Values) *url.URL {
...
 	return u
 }
-
-// appendValues appends the parameters to the url. Panics if the string is not
-// a url.
-func appendValues(u string, values ...url.Values) string {
-	up, err := url.Parse(u)
-
-	if err != nil {
-		panic(err) // should never happen
```
Comment (stevvooe) [inline registry/api/v2/urls.go]: @levigross This is never called with an invalid url. An invalid url getting through here is a programming error.

**Comment by stevvooe** [comment]
@levigross Looks fine. Please separate spelling changes and code changes into at least separate commits.

File: registry/api/v2/urls.go (hunk 1 above)
Comment (levigross) [inline registry/api/v2/urls.go]: I also didn't see this function used anywhere

File: registry/auth/htpasswd/htpasswd.go [hunk 2]
```diff
@@ -36,8 +36,8 @@ func (htpasswd *htpasswd) authenticateUser(username string, password string) err
 		return ErrAuthenticationFailure
 	}
 
-	err := bcrypt.CompareHashAndPassword([]byte(credentials), []byte(password))
```
Comment (stevvooe) [inline registry/auth/htpasswd/htpasswd.go]: Please run gofmt here. It is failing in the testsuite.

File: registry/api/v2/urls.go (hunk 1 above)
Comment (stevvooe) [inline registry/api/v2/urls.go]: @levigross I understood that. I was responding to your assertion in the PR description. I am not sure why that is. I suspect I mean to export this for use elsewhere, but we can still remove it.

**Comment by levigross** [comment]
@stevvooe I don't quite understand what I should be doing by separating the commit. Would you like me to amend the previous commit and revert  `registry/api/v2/urls.go` to it's original state and then add a separate commit removing the `appendValues` function? 

Thank you for your patience

**Comment by stevvooe** [comment]
@levigross I'd just like to have separate commits for the code changes. The change to the err return and `appendValues` are fine for a single commit. What you describe sounds find.

**Comment by levigross** [comment]
@stevvooe done

**Comment by stevvooe** [comment]
@levigross It looks like we got 4 commits, where their should be 2. Could you rebase them so the spelling changes are in one commit and the code changes are in the other? We shouldn't have a reversion commit in the history.

Sorry for the confusion.

**Pull request closed without merging**